### PR TITLE
Fetch memory info

### PR DESCRIPTION
**Pull request by farhan-sauce**
Fetch memory info

Screen Shot 2019-08-09 at 4 51 10 PM

example code
```
import Tracelib from '../src/index'
import JANK_TRACE_LOG from './__fixtures__/jankTraceLog.json'

const tasks = new Tracelib(JANK_TRACE_LOG)
const memoryInfo = tasks.getMemoryCounters()
console.log(memoryInfo)

# OUTPUT
{
  "jsHeapSizeUsed": { times: [
    289959946.271,
    289959983.488,
    289960044.53,        
  ], values: [
     22549008,
     20896736,
     21471160,
] },
  "documents": { times: [], values: [] },
  "nodes": { times: [], values: [] },
  "jsEventListeners": { times: [], values: [] },
  "gpuMemoryUsedKB": { times: [], values: [] },
}
```

### Comment
This PR should be rebased to `master` and merge only after https://github.com/saucelabs/tracelib/pull/8 gets merged.

### CODE REVIEW

File: src/timelineModel/counter/calculator.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+export default class Calculator {
+    private _zeroTime: number
+    private _minimumBoundary: number
+    private _maximumBoundary: number
+    private _workingArea: number
+
+    public constructor() {}
```
Comment (adambabik) [inline src/timelineModel/counter/calculator.ts]: Is it needed?

File: src/timelineModel/counter/calculator.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    public setZeroTime(time: number): void {
+        this._zeroTime = time
+    }
+
+    /**
+     * @override
+     * @param {number} time
+     * @return {number}
+     */
+    public computePosition(time: number): number {
```
Comment (adambabik) [inline src/timelineModel/counter/calculator.ts]: Is `time` a good name here?

File: src/timelineModel/counter/counter.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        this.times = []
+        this.values = []
+    }
+
+    /**
+     * @param {number} time
+     * @param {number} value
+     */
+    public appendSample(time: number, value: number): void {
+        if (this.values.length && this.values[this.values.length - 1] === value) return
```
Comment (adambabik) [inline src/timelineModel/counter/counter.ts]:
```suggestion
        if (this.values.slice(-1) === value) return
```

File: src/timelineModel/counter/countersGraph.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        this._gpuMemoryCounter = this._createCounter('GPU Memory')
+        this._countersByName['gpuMemoryUsedKB'] = this._gpuMemoryCounter
+    }
+
+    /**
+     * @param {?Timeline.PerformanceModel} model
+     * @param {?TimelineModel.TimelineModel.Track} track
+     */
+    public setModel(model: PerformanceModel, track: Track): CountersObject {
+        this._calculator.setZeroTime(model ? model.timelineModel().minimumRecordTime() : 0)
```
Comment (adambabik) [inline src/timelineModel/counter/countersGraph.ts]: From the caller perspective, `setZeroTime` can take a non-zero argument which is odd. I would try to rename `setZeroTime` to something different.

File: src/timelineModel/counter/countersGraph.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            return
+        }
+        const events = track.syncEvents()
+        for (let i = 0; i < events.length; ++i) {
+            const event = events[i]
+            if (event.name !== RecordType.UpdateCounters) {
+                continue
+            }
+
+            const counters: any = event.args.data
```
Comment (adambabik) [inline src/timelineModel/counter/countersGraph.ts]: Can we be more specific about `event.args.data` type?

**Review by adambabik — COMMENTED** [review]
Do you plan to add more unit tests?

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Same here, let's make sure we don't build too much custom logic into the original devtools files so the process of updating it is easier.

File: src/timelineModel/counter/counter.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
+import Calculator from './calculator'
+import { constrain, upperBound, lowerBound } from '../../utils'
+
+export default class Counter {
```
Comment (christian-bromann) [inline src/timelineModel/counter/counter.ts]: I assume this is our own class that you have implemented. Let's move these files into a dedicated `lib` or `custom` folder so we know which files are written by use and which are from the original code base

File: src/timelineModel/counter/calculator.ts (hunk 2 above)
Comment (christian-bromann) [reply src/timelineModel/counter/calculator.ts]: let's keep the naming of the original code base so we can identify the original code better

File: src/timelineModel/counter/countersGraph.ts (hunk 4 above)
Comment (christian-bromann) [reply src/timelineModel/counter/countersGraph.ts]: same, let's keep the naming as is

File: src/timelineModel/counter/counter.ts (hunk 6 above)
Comment (farhan-sauce) [reply src/timelineModel/counter/counter.ts]: `Counter` is not our own class, it's taken from [here](https://github.com/ChromeDevTools/devtools-frontend/blob/f4afd274202fc84c030cb6aeb722a811dc4ed6a2/front_end/timeline/CountersGraph.js#L255)

**Review by christian-bromann — APPROVED** [review]
👍

**Pull request merged**